cloud-sql/sqlserver/servlet/src/main/java/com/example/cloudsql/IndexServlet.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.cloudsql;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.sql.DataSource;
+
+@WebServlet(name = "Index", value = "")
+public class IndexServlet extends HttpServlet {
+
+  private static final Logger LOGGER = Logger.getLogger(IndexServlet.class.getName());
+
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse resp)
+      throws IOException, ServletException {
+    // Extract the pool from the Servlet Context, reusing the one that was created
+    // in the ContextListener when the application was started
+    DataSource pool = (DataSource) req.getServletContext().getAttribute("my-pool");
+
+    int tabCount;
+    int spaceCount;
+    List<Vote> recentVotes = new ArrayList<>();
+    try (Connection conn = pool.getConnection()) {
+      // PreparedStatements are compiled by the database immediately and executed at a later date.
+      // Most databases cache previously compiled queries, which improves efficiency.
+      PreparedStatement voteStmt =  conn.prepareStatement(
+          "SELECT TOP(5) candidate, time_cast FROM votes ORDER BY time_cast DESC");
+      // Execute the statement
+      ResultSet voteResults = voteStmt.executeQuery();
+      // Convert a ResultSet into Vote objects
+      while (voteResults.next()) {
+        String candidate = voteResults.getString(1);
+        Timestamp timeCast = voteResults.getTimestamp(2);
+        recentVotes.add(new Vote(candidate.trim(), timeCast));
```

Comment:
feel free to take or leave this suggestion - I would refactor `new Vote(...` out as a variable.